### PR TITLE
[Flask] new flask iOS target

### PR DESCRIPTION
**Pull request by owencraston**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
- so that we can have a flask variant of the mobile app
- This PR **DOES NOT** setup the proper code signing or the proper bitrise pipelines.
_2. What is the improvement/solution?_
- defined a new target, I duplicated the MetaMask target so it should have almost all of the same settings
- added a ` yarn start:ios:flask` command to the `package.json`
    - This command calls the `build.sh` file with the arguments `/scripts/build.sh ios flaskDebug"`
- Modified the `build.sh` file to handle the `flaskDebug` variant which builds the app with the scheme set to `MetaMask-Flask`
- Added the flask logo with all the different sizes. The logo can be found [here](https://www.figma.com/file/f4i6WJ1rbxtD5TUk7mypnF/Mobile-Flask-Flows?type=design&node-id=2-675&t=2YnRuCLxVHrRpoNs-0).

The new bundle id will be `io.metamask.MetaMask-Flask`
Screenshot 2023-06-14 at 10 48 28 PM

**Screenshots/Recordings**

https://github.com/MetaMask/metamask-mobile/assets/22918444/5c5781b7-ef6e-475a-997b-dfe7c528b981

Screenshot 2023-06-14 at 10 45 22 PM

### Testing/Reviewing
- checkout this branch and run `yarn setup` there should not be any major changes
- open xcode and in the top section beside the `play button`, click the MetaMask icon and a dropdown should open.
- then you should be able to select the `MetaMask-Flask` with the purple logo and click play

Screenshot 2023-06-14 at 10 36 23 PM

- the app should build and you should be able to use it as normal but the logo should be different. You should be able to build the normal MetaMask flask target alongside Flask without updating. They should be separate. 
- To test the build.sh script,  run `yarn start:ios:flask`

Screenshot 2023-06-14 at 10 42 46 PM

- the build should kick off, it takes some time but after a minute or two it should be successful

Screenshot 2023-06-14 at 10 41 52 PM

-  **NOTE**  The app will not launch in the simulator. I followed the same pattern as MetaMask-QA and noticed that that script builds but does not launch the simulator either so I think that this is alright...at least for now.
  

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/1033

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -191,11 +197,18 @@ buildIosDeviceQA(){
...
+	react-native run-ios --device --scheme "MetaMask-Flask"
+}
+
 generateArchivePackages() {
   scheme="$1"
 
   if [ "$scheme" = "MetaMask-QA" ] ; then
     exportOptionsPlist="MetaMask/IosExportOptionsMetaMaskQARelease.plist"
+  elif [ "$scheme" = "MetaMask-Flask" ] ; then
+    exportOptionsPlist="MetaMask/IosExportOptionsMetaMaskFlaskRelease.plist"
```
Comment (owencraston) [inline scripts/build.sh:211]: I am not sure what this is doing but I followed the lead of the other build variants.

**Comment by owencraston** [comment]
Here is an older attempt at creating flask build variants: https://github.com/MetaMask/metamask-mobile/pull/5151

**Comment by NicolasMassart** [comment]
> Here is an older attempt at creating flask build variants: #5151

You can probably pick most of the Android part from this old closed PR.

File: scripts/build.sh [hunk 2]
```diff
@@ -382,6 +395,12 @@ buildIos() {
 		else
 			buildIosSimulatorQA
 		fi
+	elif [ "$MODE" == "flaskDebug" ] ; then
+		if [ "$RUN_DEVICE" = true ] ; then
+			buildIosDeviceFlask
+		else
+			buildIosSimulatorFlask
+		fi
```
Comment (gantunesr) [inline scripts/build.sh:403]: Will there be a flask release mode?

File: ios/MetaMask/MetaMask-Flask-Info.plist [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	<dict>
+		<key>live</key>
+		<string>$(MM_BRANCH_KEY_LIVE)</string>
+		<key>test</key>
+		<string>$(MM_BRANCH_KEY_TEST)</string>
+	</dict>
+	<key>fox_code</key>
+	<string>$(MM_FOX_CODE)</string>
+	<key>mixpanel_token</key>
+	<string>$(MM_MIXPANEL_TOKEN)</string>
```
Comment (gantunesr) [inline ios/MetaMask/MetaMask-Flask-Info.plist:134]: Do we have a ticket to generate the new tokens?

Comment (owencraston) [reply ios/MetaMask/MetaMask-Flask-Info.plist:134]: we do: https://app.zenhub.com/workspaces/metamask-accounts-team-62505028c3853700162f65e0/issues/gh/metamask/mobile-planning/1043

File: scripts/build.sh (hunk 2 above)
Comment (owencraston) [reply scripts/build.sh:403]: Yes, I was planning on doing in this [change](https://app.zenhub.com/workspaces/metamask-accounts-team-62505028c3853700162f65e0/issues/gh/metamask/mobile-planning/1044). That way I can test the release scripts with the bitrise pipelines.

**Review by gantunesr — APPROVED** [review]

**Pull request merged**